src/main/drivers/pwm_output.c

```diff
@@ -568,6 +570,24 @@ void pwmWriteServo(uint8_t index, uint16_t value)
     }
 }
 
+#ifdef USE_DSHOT
+void changeDshotSpinRotation(dshotDirectionCommands_e directionSpin) {
```

Comment:
I suggest to do this differently. All motor interaction should happen inside `pwm_output`, there shouldn't be any external APIs calling functions that do an async motor writes.

I suggest to create a pending command structure and have an API (this function) to only update that pending command and return immediately. Actual sending data to the motors should happen inside `pwmCompleteMotorUpdate`.

This will allow us to reuse this code infrastructure in the future to implement DSHOT beeper, or in-flight motor direction change.

Reply:
I agree with @digitalentity over here. This function should only set a flag, do not call any other functions, and logic of sending the command should happen the next time when `pwmCompleteMotorUpdate` is called